pkg/dataobj/consumer/partition_processor.go

```diff
@@ -251,3 +285,31 @@ func (p *partitionProcessor) commitRecords(record *kgo.Record) error {
 	}
 	return lastErr
 }
+
+// idleFlush flushes the file if it has been idle for too long.
+// This is used to avoid holding on to memory for too long.
+// We compare the current time with the last flush time to determine if the builder has been idle.
+func (p *partitionProcessor) idleFlush() {
+	if p.builder == nil {
+		return
+	}
+
+	now := time.Now()
+	if now.Sub(p.lastFlush) < p.idleFlushTimout {
```

Comment:
You could also use time.Since here: `time.Since(p.lastFlush) < p.idleFlushTimeout`
In practice it makes very little difference but its slightly easier to read